### PR TITLE
Demeter-V2 Changes

### PR DESCRIPTION
**Pull request by YashP16**
### New Features: 
* Add Farm plugins:
  * Expirable farms: Farms are created with an initial expiry of 100 days. Farm admins can later on extend farms for a standard fee.
  * Operable farms: Allows users to increase/decrease liquidity while being staked in the farm.
  * Active liquidity farms
  * Fixed APR farms
* Make contracts chain agnostic. 
* Add support for new DEXs: 
   *   Add farms for CamelotV3, Sushiswap, E20-token farms (ex: Balancer, Aave)
* Contracts are audited by QuantStamp: [Report](https://github.com/Sperax/Demeter-Protocol/blob/dev/audits/Quantstamp/SperaxFarmsAuditReport.pdf)

### CODE REVIEW

File: contracts/utils/TokenUtils.sol
Comment (parv3213) [inline contracts/utils/TokenUtils.sol:1]: Functions in this library was made public, right?

File: pip-requirements.txt
Comment (parv3213) [inline pip-requirements.txt:1]: This may clutter the repo. I suggest to delete.

File: package.json [hunk 1]
```diff
@@ -7,19 +7,31 @@
     "test": "tests"
   },
   "scripts": {
-    "install-pip-packages": "pip3 install -r requirements.txt",
-    "initialize-pre-commit": "pre-commit install -c ./dev-utils/.pre-commit-config.yaml",
-    "prepare": "npm-run-all install-pip-packages initialize-pre-commit",
-    "solhint": "./node_modules/.bin/solhint -f table -c ./dev-utils/.solhint.json",
-    "prettier:solidity": "./node_modules/.bin/prettier --write **/*.sol"
+    "install-pip-packages": "pip install -r pip-requirements.txt",
```
Comment (parv3213) [inline package.json]: This may clutter the dependencies. I suggest to remove.

Comment (parv3213) [reply package.json]: If removing, also remove this from `prepare`

File: remappings.txt
Comment (parv3213) [inline remappings.txt:1]: This should not be required. As we use `libs = ["node_modules", "lib"]` in foundry.toml

File: slither.config.json
Comment (parv3213) [inline slither.config.json:1]: Suggest to delete this file.

**Review by parv3213 — APPROVED** [review]
Approving with comments for improvements.

File: pip-requirements.txt
Comment (arcantheon) [reply pip-requirements.txt:1]: Removed: b0ceeb2856a5c9a1022b6145a3d486301d4d6e92

File: contracts/utils/TokenUtils.sol
Comment (arcantheon) [reply contracts/utils/TokenUtils.sol:1]: Earlier they were public now they are internal.

File: package.json (hunk 1 above)
Comment (arcantheon) [reply package.json]: Done: b0ceeb2856a5c9a1022b6145a3d486301d4d6e92

File: remappings.txt
Comment (arcantheon) [reply remappings.txt:1]: Done: 0x14D99412dAB1878dC01Fe7a1664cdE85896e8E50

File: slither.config.json
Comment (arcantheon) [reply slither.config.json:1]: Done: 0x14D99412dAB1878dC01Fe7a1664cdE85896e8E50

File: pip-requirements.txt
Comment (parv3213) [reply pip-requirements.txt:1]: We can also remove `npm-run-all`, I guess

Comment (arcantheon) [reply pip-requirements.txt:1]: Removed: https://github.com/Sperax/Demeter-Protocol/commit/564e61150343f5ba6e6905180b9b620050c9869d

**Review by parv3213 — APPROVED** [review]

**Pull request merged**